### PR TITLE
Fixes for (cross)compiling with GCC 2.95.3 and bugs found

### PR DESCRIPTION
**Pull request by adamel**
Still compiles and works on SAS/C.

### CODE REVIEW

File: Nodes/NiKomFuncs.h [hunk 1]
```diff
@@ -1,55 +1,20 @@
 #include <stdio.h>
 #include <devices/serial.h>
+#include <dos/dos.h>
+
+#include <NiKomStr.h>
+#include <NiKomNodeCommon.h>
 /*
  * TODO: The long time goal is for this file to die. Every .c
```
Comment (punktniklas) [inline Nodes/NiKomFuncs.h]: Can you add this TODO to the new NiKomNodeCommon.h as well.

File: InitNiKom/InitNiKom.c [hunk 2]
```diff
@@ -28,7 +28,7 @@ void createDirOrDie(char *dirname) {
   BPTR lock;
   struct FileInfoBlock __aligned fib;
   
-  if(lock = Lock(dirname, ACCESS_READ)) {
+  if((lock = Lock(dirname, ACCESS_READ))) {
```
Comment (punktniklas) [inline InitNiKom/InitNiKom.c]: Why does gcc want these extra parens..? It's not like there is any ambiguity without them.

File: Nodes/Terminal.c [hunk 3]
```diff
@@ -99,7 +99,7 @@ int GetChar(void) {
 }
 
 int handle1bChar(void) {
-  unsigned char ch;
+  char ch;
```
Comment (punktniklas) [inline Nodes/Terminal.c]: Why removing "unsigned" here and not in the function below?

File: InitNiKom/InitNiKom.c (hunk 2 above)
Comment (adamel) [inline InitNiKom/InitNiKom.c]: To be able to tell intentional assignments from bugs like the one fixed commit 3f0646a. :-)
Those bugs really happen, and it's nice for the compiler to be able to catch them, and the extra parentheses is simply a way to tell the compiler "I really know what I'm doing" so false warnings do not hide the real ones.

File: Nodes/Terminal.c (hunk 3 above)
Comment (adamel) [inline Nodes/Terminal.c]: It's was not needed as there is no comparison with constants > 127 there, but I'll update them anyway for consistency.

**Comment by adamel** [comment]
Updated. Now `proto/alib.h` is included where needed when compiling with gcc. On SAS/C `proto/exec.h`   includes `clib/alib_protos.h` but with gcc you can only get it either by including it directly or by including `proto/alib.h`.

File: InitNiKom/InitNiKom.c (hunk 2 above)
Comment (punktniklas) [inline InitNiKom/InitNiKom.c]: So the rule in gcc is "if there is an extra set of parens it won't generate a warning if there is an assignment in the if condition"?
The problem with this is that there is no such trick in SAS/C so anyone developing with SAS/C is at risk of introducing warnings in gcc without realizing it. But I guess it's not a huge problem..

File: Nodes/ConsoleIO.c [hunk 4]
```diff
@@ -1,5 +1,7 @@
 #include <exec/types.h>
 #include <proto/exec.h>
+#include <clib/alib_protos.h>
```
Comment (punktniklas) [inline Nodes/ConsoleIO.c]: Here's another clib include.

File: InitNiKom/InitNiKom.c (hunk 2 above)
Comment (punktniklas) [inline InitNiKom/InitNiKom.c]: Then again.. I guess that's true for many of the other updates as well. (e.g. %d -> %ld)

Comment (adamel) [inline InitNiKom/InitNiKom.c]: Well, I guess it's possible to setup something like https://travis-ci.org/ to automatically make test builds using gcc when new code is pushed.

File: Nodes/ConsoleIO.c (hunk 4 above)
Comment (adamel) [inline Nodes/ConsoleIO.c]: Ah, fixed.

**Pull request merged**